### PR TITLE
added async queries

### PR DESCRIPTION
**Pull request by aflegel**
had to modify the pool ids query as async functions cannot have an out variable

### CODE REVIEW

File: DataFramework/Context/ProbabilityContext.Accessors.cs [hunk 1]
```diff
@@ -13,17 +15,17 @@ public static class ProbabilityContextAccessors
 		/// </summary>
 		/// <param name="searchForPool"></param>
 		/// <returns></returns>
-		public static int? GetPoolId(ProbabilityContext context, List<PoolDie> searchForPool) =>
-			searchForPool.Select(die => context.PoolDice.Where(w => w.DieId == die.DieId && w.Quantity == die.Quantity && w.Pool.PoolDice.Count == searchForPool.Count())
-			.Select(s => s.PoolId)).Aggregate((result, next) => result.Intersect(next)).FirstOrDefault();
+		public static async Task<int?> GetPoolId(ProbabilityContext context, List<PoolDie> searchForPool) =>
+			await searchForPool.Select(die => context.PoolDice.Where(w => w.DieId == die.DieId && w.Quantity == die.Quantity && w.Pool.PoolDice.Count == searchForPool.Count())
+			.Select(s => s.PoolId)).Aggregate((result, next) => result.Intersect(next)).FirstOrDefaultAsync();
```
Comment (ChrisDusyk) [inline DataFramework/Context/ProbabilityContext.Accessors.cs:20]: Are you over-awaiting here?
```suggestion
		public static Task<int?> GetPoolId(ProbabilityContext context, List<PoolDie> searchForPool) =>
			searchForPool.Select(die => context.PoolDice.Where(w => w.DieId == die.DieId && w.Quantity == die.Quantity && w.Pool.PoolDice.Count == searchForPool.Count())
			.Select(s => s.PoolId)).Aggregate((result, next) => result.Intersect(next)).FirstOrDefaultAsync();
```

File: DataFramework/Context/ProbabilityContext.Accessors.cs [hunk 2]
```diff
@@ -13,17 +15,17 @@ public static class ProbabilityContextAccessors
...
 
 		/// <summary>
 		/// Tries to find the pool id from a given name
 		/// </summary>
 		/// <param name="context"></param>
 		/// <param name="poolName"></param>
 		/// <returns></returns>
-		public static int? GetPoolIdByName(this ProbabilityContext context, string poolName) => context.Pools.Where(w => w.Name == poolName).Select(s => s.PoolId).FirstOrDefault();
+		public static async Task<int?> GetPoolIdByName(this ProbabilityContext context, string poolName) => await context.Pools.Where(w => w.Name == poolName).Select(s => s.PoolId).FirstOrDefaultAsync();
```
Comment (ChrisDusyk) [inline DataFramework/Context/ProbabilityContext.Accessors.cs:28]: Same question for over-awaiting

**Review by ChrisDusyk — APPROVED** [review]

File: DataFramework/Context/ProbabilityContext.Accessors.cs (hunk 1 above)
Comment (aflegel) [reply DataFramework/Context/ProbabilityContext.Accessors.cs:20]: You'd think but the compiler would complain about return types if I tried to pass the task down directly.  `FirstOrDefaultAsync` doesn't return a nullable integer.

Comment (ChrisDusyk) [reply DataFramework/Context/ProbabilityContext.Accessors.cs:20]: Good to know! Then yea, everything looks good.

**Pull request merged**